api/logical.go

```diff
@@ -96,6 +117,34 @@ func (c *Logical) Delete(path string) (*Secret, error) {
 }
 
 func (c *Logical) Unwrap(wrappingToken string) (*Secret, error) {
+	var data map[string]interface{}
+	if wrappingToken != "" {
```

Comment:
Actually, even if it *is* empty we always use the new method -- it'll just use the client's token instead of an alternate. But you're right -- in the old method, if the wrapping token is empty, we should not set the client to an empty token. Fixed!